File: mobile/src/services/firebase/authService.ts
import {
  createUserWithEmailAndPassword,
  onAuthStateChanged,
  sendPasswordResetEmail,
  signInWithEmailAndPassword,
  signOut,
  User,
} from "firebase/auth";
import { auth } from "./firebaseConfig";

export function subscribeToAuthChanges(callback: (user: User | null) => void): () => void {
  return onAuthStateChanged(auth, callback);
}

export async function login(email: string, password: string): Promise<User> {
  const credential = await signInWithEmailAndPassword(auth, email, password);
  return credential.user;
}

export async function register(email: string, password: string): Promise<User> {
  const credential = await createUserWithEmailAndPassword(auth, email, password);
  return credential.user;
}

export async function logout(): Promise<void> {
  await signOut(auth);
}

export async function resetPassword(email: string): Promise<void> {
  await sendPasswordResetEmail(auth, email);
}

File: ios-mobile/ios-mobile/Services/Auth/Services/AuthService.swift
import Foundation

protocol AuthService {
    var authStateChanges: AsyncStream<AppUser?> { get }

    func login(email: String, password: String) async throws
    func register(email: String, password: String) async throws
    func logout() throws
    func resetPassword(email: String) async throws
    func deleteAccount() async throws
}
